### PR TITLE
[jest-axe] Update tsconfig

### PR DESCRIPTION
**Pull request by wsmd**
Hi, `@types/jest-axe` maintainers (@JoshuaKGoldberg)!

I'm currently running into an issue in #29753 where the build is failing because of this error (which I'm also getting locally):

```

=== ERRORS ===
Error in jest-axe
Error: /home/travis/build/DefinitelyTyped/DefinitelyTyped/types/jest-axe/node_modules/axe-core/axe.d.ts:23:24
ERROR: 23:24  expect  TypeScript@next compile error: 
Cannot find name 'Node'.
    at /home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/dtslint/bin/index.js:158:19
    at Generator.next (<anonymous>)
    at fulfilled (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/dtslint/bin/index.js:5:58)
    at <anonymous>
Error: The following packages had errors: jest-axe
    at runTests (/home/travis/build/DefinitelyTyped/DefinitelyTyped/node_modules/types-publisher/src/tester/test-runner.ts:126:8)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:189:7)
npm ERR! Test failed.  See above for more details.
```

https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/442340557?utm_source=github_status&utm_medium=notification

It looks like `axe-core` is a dependency of `@types/jest-axe`, and the declaration file for `axe-code`[ is referencing `Node`](https://github.com/dequelabs/axe-core/blob/f98b83085afb79d5327ea73b0f40ac2c9e346cc1/axe.d.ts#L23
), which is available via the "dom" library. By making this change, I was able to fix issue and the tests for my other PR are passing.

Please let me know if this makes sense to you. I'm also curious to know why this wasn't an issue before 🤔.

### CODE REVIEW

**Review by JoshuaKGoldberg — APPROVED** [review]

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
Heya @wsmd, sorry about the delay! 

+1 to fixing the build break but -1 to including the `dom` lib in `jest-axe`'s typings. Doing so would imply the library has access to the types globally declared in DOM environment (browsers), which is not true for the typical execution of the library (Node). Per https://github.com/nickcolley/jest-axe#usage the library is intended to be used in Node.

Instead of adding to the tsconfig, it seems like it'd be better to:
* Add in a stub interface for `Node`: something like `interface Node {}`
* File an issue on the `axe-core` typings to provide their own `Node` if it doesn't exist

This may seem like bikeshedding, but it's actually relevant for some projects: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/24419 https://github.com/DefinitelyTyped/DefinitelyTyped/issues/21517
https://github.com/Microsoft/TSJS-lib-generator/issues/327

_(Edit: whoops, didn't mean to approve changes 5 minutes ago...)_

**Comment by wsmd** [comment]
Thanks for taking a look, and for the detailed explanation, @JoshuaKGoldberg! That makes a ton of sense actually. The examples are very helpful too.

I'll try resolving this issue directly with `axe-core`, if no progress is made very soon, I'll go ahead and stub `Node`  in `jest-axe`'s typings since it's currently blocking new PRs to `@types/jest` from getting merged.

**Comment by wsmd** [comment]
> File an issue on the axe-core typings to provide their own Node if it doesn't exist

After some digging into `axe-core`, I realized that they actually may be depending on some `Node`/`document`-like objects in their code. So it kinda makes sense for them to rely on the `dom` lib. I filed an issue already on their repo. In the meantime, would you be open stubbing `Node` in `jest-axe`? I do like this approach more now that I have given it some thought.

Let me know what you think.

**Review by JoshuaKGoldberg — APPROVED** [review]

File: types/jest-axe/index.d.ts [hunk 1]
```diff
@@ -8,6 +8,10 @@
 
 import { AxeResults, Result, RunOnly } from "axe-core";
 
+declare module "axe-core" {
```
Comment (rbuckton) [inline types/jest-axe/index.d.ts]: Does `"axe-core"` have its own `Node` type? If you are stubbing the definition for `Node` from the DOM, then this should be in a global declaration:
```suggestion
declare global {
```

**Review by rbuckton — CHANGES_REQUESTED** [review]

File: types/jest-axe/index.d.ts (hunk 1 above)
Comment (wsmd) [reply types/jest-axe/index.d.ts]: `axe-core` doesn't have it's own `Node` type. However, by doing so, wouldn't `jest-axe` stub that for all of its dependents? We don't want that, right?

Comment (rbuckton) [reply types/jest-axe/index.d.ts]: If `"axe-core"` actually depends on a global `Node` and you create a forward declaration for `Node` _inside_ of `"axe-core"`, then it won't be for the actual global `Node` and will always be an empty interface (meaning just about anything would be assignment compatible to it).

If it is a forward declaration for a global type, it should be declared as a global. This is similar to what the NodeJS definitions currently do for forward declarations for other global types like `Iterable`.

Comment (wsmd) [reply types/jest-axe/index.d.ts]: Makes sense!

Thanks for the explanation @rbuckton 👍

**Review by rbuckton — APPROVED** [review]

**Pull request merged**